backend/tests/bdd_samples/test_crcelements_bdd_style.py

```diff
@@ -0,0 +1,129 @@
+@hp.example(crc=b'\x00\x00\x00\x00')
+@hp.example(crc=b'\xff\xff\xff\xff')
+@hp.example(crc=b'\xff\xff\xff\xff' + b'\x00\x00\x00\x01')
+def test_crcelement_init_crc(crc: bytes) -> None:
+    """
+    Scenario: CRC of data is generated/parsed from the input.
+    Given: We have CRCElement object.
+    And: The CRC generated/parsed is in bytes.
+    When: The CRC is between \\x00\\x00\\x00\\x00 and \\xff\\xff\\xff\\xff'
+    And: It's length is 4.
+    Then: ValueError should not be raised.
+    """
+    if (b'\x00\x00\x00\x00' <= crc <= b'\xff\xff\xff\xff') and (len(crc) == 4):
+        crcelements.CRCElement(crc=crc)
+    else:
+        with pt.raises(ValueError):
+            crcelements.CRCElement(crc=crc)
+
+
+@pt.mark.parametrize('crc,payload,_', example_crc_good)
+def test_crcelement_integrity_pass(
+        crc: bytes, payload: bytes, _: bytes
+) -> None:
+    """
+    Scenario: The system reads file/processes message from mcu/frontend.
+        The message has been parsed 
+    Given: We have CRCElement object.
+    And: The parsed CRC and payload, in bytes.
+    When: The parsed CRC matches the computed CRC
+    Then: ProtocolDataError should not be raised.
+    """
+    # Given: We have CRCElement object.
+    crc_element = crcelements.CRCElement()
+
+    # And: The parsed CRC and payload, in bytes.
+    crc_element.crc = crc
+    crc_element.payload = payload
+    
+    try:
+        # When: The parsed CRC matches the computed CRC
+        crc_element.check_integrity()
+    
+    # Then: ProtocolDataError should not be raised.
+    except exceptions.ProtocolDataError as err:
+        pt.fail("Integrity check failed: {0}".format(err))
+
+
+@pt.mark.parametrize('crc,payload,body', example_crc)
+def test_crcelement_parse(
+        crc: bytes, payload: bytes, body: bytes
+) -> None:
+    """
+    Scenario: The system reads file/processes message from mcu/frontend.
+    Given: A protobuf message in bytes.
+    And: A CRC element.
+    When: The protobuf message has correct size
+    Then: We should get seperated CRC key and message body.
+    """
+    # Given: A protobuf message in bytes.
+    # And: A CRC element.
+    crc_element = crcelements.CRCElement()
+    
+    # When: The protobuf message has correct size
+    crc_element.parse(body)
+    
+    # Then: We should get seperated CRC key and message body.
+    assert crc_element.crc == crc
+    assert crc_element.payload == payload
+
+
+@pt.mark.parametrize('_,__,body', example_crc_bad)
+def test_crcelements_rx_invalid_crc(
+        _: bytes, __: bytes, body: bytes
+) -> None:
+    """
+    Scenario: The system reads corrupted file or
+        processes corrupted message.
+    Given: A protobuf message in bytes.
+    And: A CRC message receiver.
+    When: The message bytes are corrupted
+    Then: Raise ProtocolDataError for that message.
+    """
+    # Given: A protobuf message in bytes.
+    # And: A CRC message receiver.
+    receiver = crcelements.CRCReceiver()
```

Comment:
I think line 122 makes more sense as just "Given: a CRC message receiver", and line 125 makes more sense as just "When: the receiver is given a body with corrupted bytes", and line 128-129 might be more specific as "Then: processing the body raises a ProtocolDataError"

Reply:
Agreed